pkg/notes/template.go

```diff
@@ -0,0 +1,250 @@
+	"api-machinery":     {"API Machinery", 0},
+	"apps":              {"Apps", 1},
+	"auth":              {"Auth", 2},
+	"autoscaling":       {"Autoscaling", 3},
+	"cli":               {"CLI", 4},
+	"cloud-provider":    {"Cloud Provider", 5},
+	"cluster-lifecycle": {"Cluster Lifecycle", 6},
+	"instrumentation":   {"Instrumentation", 7},
+	"network":           {"Network", 8},
+	"node":              {"Node", 9},
+	"release":           {"Release", 10},
+	"scheduling":        {"Scheduling", 11},
+	"storage":           {"Storage", 12},
+	"windows":           {"Windows", 13},
+}
+
+type definedSIG struct {
+	name string
+	id   int
+}
+
+type Metrics struct {
+	Changes    []string
+	Added      []string
+	Removed    []string
+	Deprecated []string
+}
+
+type Features struct {
+	Stable              []string
+	Beta                []string
+	Alpha               []string
+	StagingRepositories []string
+	CliImprovements     []string
+	Misc                []string
+}
+
+const notesTemplate = `## What’s New (Major Themes)
+
+<!-- Add themes from Comms Blog here -->
+
+## Known Issues
+
+<!-- Add issues from known issues bucket (known-issues-bucket.Md) here -->
+
+## Urgent Upgrade Notes
+
+### (No, really, you MUST read this before you upgrade)
+{{range .UpgradeNotes}}{{if .Header}}
+#### {{.Header}}
+
+{{range .Content}}{{.}}
+{{end}}{{end}}{{end}}
+## Deprecations and Removals
+
+<!--
+Add it in the format
+- Component
+  - deprecation
+  - removal
+-->
+
+## Metrics Changes
+
+{{range .Metrics.Changes}}{{.}}
+{{end}}
+### Added metrics
+
+{{range .Metrics.Added}}{{.}}
+{{end}}
+### Removed metrics
+
+{{range .Metrics.Removed}}{{.}}
+{{end}}
+### Deprecated/changed metrics
+
+{{range .Metrics.Deprecated}}{{.}}
+{{end}}
+## Notable Features
+
+### Stable
+
+{{range .Features.Stable}}{{.}}
+{{end}}
+### Beta
+
+{{range .Features.Beta}}{{.}}
+{{end}}
+### Alpha
+
+{{range .Features.Alpha}}{{.}}
+{{end}}
+### Staging Repositories
+
+{{range .Features.StagingRepositories}}{{.}}
+{{end}}
+### CLI Improvements
+
+{{range .Features.CliImprovements}}{{.}}
+{{end}}
+### Misc
+
+{{range .Features.Misc}}{{.}}
+{{end}}
+## API Changes
+
+{{range .APIChanges}}{{.}}
+{{end}}
+## Other notable changes
+{{range .Other}}{{if .Header}}
+### {{ .Header }}
+
+{{range .Content}}{{.}}
+{{end}}{{end}}{{end}}
+## Dependencies
+
+<!-- Add by hand -->
+`
+
+func CreateFromTemplate(logger log.Logger, notes ReleaseNotes) (*Template, error) {
+	res := &Template{}
+	for _, note := range notes {
+		for _, sig := range note.SIGs {
+
+			definedSIG, ok := availableSIGs[sig]
+			if !ok {
+				level.Warn(logger).Log("msg", "unable to find ID for SIG "+sig)
+				continue
+			}
+			i := definedSIG.id
+
+			if note.ActionRequired {
+				res.UpgradeNotes[i].setHeaderIfNeeded(definedSIG.name)
+				res.UpgradeNotes[i].Content = markdownAppend(res.UpgradeNotes[i].Content, note)
+
+			} else if wordIn(note, "metrics") {
+				res.Metrics.classifyMetricsAppend(note)
+
+			} else if wordIn(note, "api") {
```

Comment:
This might catch PRs against the apiserver which is not (AFAIK) classified as an API change. Is this also how the bash script tackles it?